plutus-core/plutus-ir/test/PlutusIR/Generators/QuickCheck/Tests.hs

```diff
@@ -184,3 +194,22 @@ prop_noTermShrinkLoops = withMaxSuccess 10 $
   forAllDoc "ty,tm" genTypeAndTerm_
     (\(ty', tm') -> filter ((/= tm') . snd) $ shrinkClosedTypedTerm (ty', tm')) $ \(ty, tm) ->
   tm `notElem` map snd (shrinkClosedTypedTerm (ty, tm))
+
+-- | Check that evaluation of the given term doesn't fail with a structural error.
+noStructuralErrors :: UPLC.Term Name DefaultUni DefaultFun () -> IO ()
+noStructuralErrors term =
+  void . evaluate . unsafeToEvaluationResult . fst $ do
+    let budgeting = restricting . ExRestrictingBudget $ ExBudget 1000000000 1000000000
```

Comment:
Can you use [restrictingEnormous](https://github.com/IntersectMBO/plutus/blob/200ad0ffe59f841b6cae82995d5f080ba71aab0d/plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/ExBudgetMode.hs#L157) here?